schemas/participant.ts

```diff
@@ -1,40 +1,40 @@
 import { z } from 'zod';
 
+// Utility function to check for non-whitespace characters
+const hasNonWhitespaceCharacters = (input: string | undefined) =>
+  input ? input.trim() !== '' : true;
+
 export const participantIdentifierSchema = z
   .string()
+  .trim()
   .min(1, { message: 'Identifier cannot be empty' })
   .max(255, { message: 'Identifier too long. Maximum of 255 characters.' })
-  // Ensure that the identifier is not just whitespace or have leading/trailing spaces
-  .refine((identifier) => identifier.trim() === identifier, {
-    message: 'Identifier cannot have leading or trailing spaces',
+  .refine(hasNonWhitespaceCharacters, {
+    message: 'Identifier requires one or more non-whitespace characters.',
```

Comment:
Won't calling `trim()` first mean that a string consisting of a space will have that space removed, so it will then fail with "identifier cannot be empty"? I think you want to call trim at the end.

Given we have vitest running now, why not write some tests for this parser? It will give you some practice, and also allow you to cover all the scenarios.

Reply:
Thanks for catching this. I moved `trim()` after min checks but before `refine`. This simplifies the `hasNonWhitespaceCharacters` utility function, since the input is already trimmed.

I added extensive tests to cover all the scenarios for each schema, rows, and lists of rows. All passing. 

Refactored `participantIdentifierOptionalSchema` and `participantLabel` schema to use `trim() `and to not use the `hasNonWhitespaceCharacters`. Since these are optional, they should be trimmed and pass with whitespace or empty strings. These are then entered into the db either as undefined for label or assigned an id for identifier.

[7651da4](https://github.com/complexdatacollective/Fresco/pull/140/commits/7651da470f9b03c789d5b0fd680a90872bfda309) & [e1a0507](https://github.com/complexdatacollective/Fresco/pull/140/commits/e1a0507cdc653ca84c79c8ded34645042c50e3ae)